cli/storage_info.c

```diff
@@ -98,44 +102,54 @@ hse_storage_info(const char *const kvdb_home)
                 cJSON_GetObjectItemCaseSensitive(root, "allocated_bytes"));
             data->mi_used_bytes = cJSON_GetNumberValue(
                 cJSON_GetObjectItemCaseSensitive(root, "used_bytes"));
+
+            if (data->mi_allocated_bytes > 0)
+                mc_present[i] = true;
         }
     } else {
+        if (hse_err_to_errno(err) == ENOENT)
+            fprintf(stderr, "No such KVDB (%s)\n", kvdb_home);
         return hse_err_to_errno(err);
     }
 
+    rowid = 0;
     for (int i = HSE_MCLASS_BASE; i < HSE_MCLASS_COUNT; i++) {
-        const int base = i * NELEM(headers);
-        values[base] = hse_mclass_name_get(i);
+        if (mc_present[i]) {
+            const int base = rowid * NELEM(headers);
+            values[base] = hse_mclass_name_get(i);
 
-        rc = snprintf(nums[i][0], sizeof(nums[i][0]), "%lu", info[i].mi_allocated_bytes);
-        if (rc < 0) {
-            rc = EBADMSG;
-            goto out;
-        } else if (rc >= sizeof(nums[i][0])) {
-            rc = EMSGSIZE;
-            goto out;
-        } else {
-            rc = 0;
-        }
-        values[base + 1] = nums[i][0];
+            rc = snprintf(nums[rowid][0], sizeof(nums[rowid][0]), "%lu",
+                          info[i].mi_allocated_bytes);
+            if (rc < 0) {
+                rc = EBADMSG;
+                goto out;
+            } else if (rc >= sizeof(nums[rowid][0])) {
+                rc = EMSGSIZE;
+                goto out;
+            } else {
+                rc = 0;
+            }
+            values[base + 1] = nums[rowid][0];
 
-        rc = snprintf(nums[i][1], sizeof(nums[i][1]), "%lu", info[i].mi_used_bytes);
-        if (rc < 0) {
-            rc = EBADMSG;
-            goto out;
-        } else if (rc >= sizeof(nums[i][0])) {
-            rc = EMSGSIZE;
-            goto out;
-        } else {
-            rc = 0;
-        }
-        values[base + 2] = nums[i][1];
+            rc = snprintf(nums[rowid][1], sizeof(nums[rowid][1]), "%lu", info[i].mi_used_bytes);
+            if (rc < 0) {
+                rc = EBADMSG;
+                goto out;
+            } else if (rc >= sizeof(nums[rowid][1])) {
+                rc = EMSGSIZE;
+                goto out;
+            } else {
+                rc = 0;
+            }
+            values[base + 2] = nums[rowid][1];
 
-        values[base + 3] = info[i].mi_path;
+            values[base + 3] = info[i].mi_path;
+
+            rowid++;
+        }
     }
```

Comment:
this is madness. why not printf with fixed width fields?

Reply:
What is this comment in reference to? The previous code?

Reply:
Yes.  So much code to print a table when all but the last columns have known max-size widths.  Could simply use "%-15s  %15d  %15d  %-s".

Reply:
The goal of tprint() was to be generically useful to support a table mode output. I seem to remember you approving the original PR :).